src/IO/GeoJSONReader.php

```diff
@@ -0,0 +1,327 @@
+            throw GeometryIOException::invalidGeoJSON('Missing or Malformed "Feature.type" attribute.');
+        }
+
+        // Verify Geometry exists and is array
+        if (! isset($feature['geometry']) || ! is_array($feature['geometry'])) {
+            throw GeometryIOException::invalidGeoJSON('Missing or Malformed "Feature.geometry" attribute.');
+        }
+
+        return $this->readGeometry($feature['geometry']);
+    }
+
+    /**
+     * @param array $geometry
+     *
+     * @return Geometry
+     *
+     * @throws GeometryIOException
+     * @throws \Brick\Geo\Exception\CoordinateSystemException
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     * @throws \Brick\Geo\Exception\UnexpectedGeometryException
+     */
+    protected function readGeometry(array $geometry) : Geometry
+    {
+        // Verify Geometry TYPE
+        if (! isset($geometry['type']) || ! is_string($geometry['type'])) {
+            throw GeometryIOException::invalidGeoJSON('Missing or Malformed "Geometry.type" attribute.');
+        }
+
+        $geoType = $geometry['type'];
+
+        // Verify Geometry COORDINATES
+        if (! isset($geometry['coordinates']) || ! array($geometry['coordinates'])) {
+            throw GeometryIOException::invalidGeoJSON('Missing or Malformed "Geometry.coordinates" attribute.');
+        }
+
+        $geoCoords = $geometry['coordinates'];
+
+        $hasZ = $this->hasZ($geoCoords);
+        $hasM = false;
+        $srid = 4326;
+
+        $cs = new CoordinateSystem($hasZ, $hasM, $srid);
+
+        switch ($geoType) {
+            case 'Point':
+                return $this->genPoint($cs, $geoCoords);
+
+            case 'MultiPoint':
+                return $this->genMultiPoint($cs, $geoCoords);
+
+            case 'LineString':
+                return $this->genLineString($cs, $geoCoords);
+
+            case 'MultiLineString':
+                return $this->genMultiLineString($cs, $geoCoords);
+
+            case 'Polygon':
+                return $this->genPolygon($cs, $geoCoords);
+
+            case 'MultiPolygon':
+                return $this->genMultiPolygon($cs, $geoCoords);
+        }
+
+        throw GeometryIOException::unsupportedGeoJSONType($geoType);
+    }
+
+    /**
+     * [x, y]
+     *
+     * @param CoordinateSystem $cs
+     * @param array            $coords
+     *
+     * @return Point
+     *
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     */
+    private function genPoint(CoordinateSystem $cs, array $coords) : Point
+    {
+        return new Point($cs, ...$coords);
+    }
+
+    /**
+     * [[x, y], ...]
+     *
+     * @param CoordinateSystem $cs
+     * @param array            $coords
+     *
+     * @return MultiPoint
+     *
+     * @throws \Brick\Geo\Exception\CoordinateSystemException
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     * @throws \Brick\Geo\Exception\UnexpectedGeometryException
```

Comment:
Same comment as above, can be grouped into `@throws GeometryException`; please apply this to all gen*() methods below.